### DISCUSSION TITLE
v3版本的cloudflare workers中转

### DISCUSSION THREAD

**Discussion by fucktimi** [AlistGo/alist#1715] [Q&A] [answered]
复制了v3版本的js代码，用了跟v2一样的address跟token，但是下载onedrive只有一个错误页面{"code":401,"message":"sign mismatch"}？
v2版本是正常使用的。

**Comment by DarremMolko** [comment, marked as answer]
The token from v2 doesn't work with v3. You can find the correct token in Settings > Other
![IMG_20220918_213616_406](https://user-images.githubusercontent.com/70179709/190934916-5f73b78f-4149-4ef6-b6aa-a980e8747963.jpg)

**Reply by fucktimi** [reply]
Thanks

**Reply by rakaki** [reply]
请教下，workers不是被墙了吗，还能怎么用吗

**Reply by fucktimi** [reply]
> 请教下，workers不是被墙了吗，还能怎么用吗

直接works是不能了，被sni阻断了，但是可以申请自己的域名，把自己的域名添加到cloudflare，就可以创建一个到workers的路由，访问自己域名就是访问workers。教程网上大把。

**Reply by rakaki** [reply]
你好，再请教下。我是在自己的物理小主机上部署的alist，然后用cloudflared做了内网穿透。然后用workers。然后访问alist的onedrive，看日志会出现canceled的错，你有遇到过吗？

  "outcome": "canceled",
  "scriptName": "alist",
  "exceptions": [],
  "logs": [],
  "eventTimestamp": 1664366083185,
  "event": {
    "request": {
      "url": "https://alist.domain.dm/ODmy/%E3%80%8A%E6%AD%A6%E6%9E%97%E5%A4%96%E4%BC%A0%E3%80%8B%E6%9C%AA%E5%88%A0%E8%8A%82%E5%8F%B0%E6%B9%BE%E6%B8%85%E6%99%B0%E7%89%88/[%E6%AD%A6%E6%9E%97%E5%A4%96%E4%BC%A0].02.%E6%8E%8C%E6%9F%9C%E7%9A%84%E6%85%B0%E9%97%AE%E8%A3%85%E7%97%85%E8%80%85.%E8%B7%91%E5%A0%82%E4%BA%BA%E6%94%B6%E6%8B%BE%E5%AF%BB%E8%A1%85%E5%A5%B3.avi?sign=REDACTED=:0",
      "method": "GET",
      "headers": {
        "accept": "*/*",
        "accept-encoding": "gzip",
        "accept-language": "zh-CN,zh;q=0.9,en;q=0.8,en-GB;q=0.7,en-US;q=0.6",
        "cf-connecting-ip": "50.7.252.66",
        "cf-ipcountry": "SG",
        "cf-ray": "751c4933ea818820",
        "cf-visitor": "{\"scheme\":\"https\"}",
        "connection": "Keep-Alive",
        "host": "alist.domain.dm",
        "if-range": "\"{09D21B07-8BD1-451B-8AB8-17665C861D77},5\"",
        "range": "bytes=1638400-525832191",
        "sec-ch-ua": "\"Microsoft Edge\";v=\"105\", \"Not)A;Brand\";v=\"8\", \"Chromium\";v=\"105\"",
        "sec-ch-ua-mobile": "?0",
        "sec-ch-ua-platform": "\"Windows\"",
        "sec-fetch-dest": "video",
        "sec-fetch-mode": "no-cors",
        "sec-fetch-site": "cross-site",
        "user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36 Edg/105.0.1343.53",
        "x-forwarded-proto": "https",
        "x-real-ip": "50.7.252.66"
      },
      "cf": {
        "clientTcpRtt": 1,
        "longitude": "103.78360",
        "latitude": "1.29820",
        "tlsCipher": "AEAD-AES128-GCM-SHA256",
        "continent": "AS",
        "asn": 30058,
        "clientAcceptEncoding": "identity;q=1, *;q=0",
        "country": "SG",
        "tlsClientAuth": {
          "certIssuerDNLegacy": "",
          "certIssuerSKI": "",
          "certSubjectDNRFC2253": "",
          "certSubjectDNLegacy": "",
          "certFingerprintSHA256": "",
          "certNotBefore": "",
          "certSKI": "",
          "certSerial": "",
          "certIssuerDN": "",
          "certVerified": "NONE",
          "certNotAfter": "",
          "certSubjectDN": "",
          "certPresented": "0",
          "certRevoked": "0",
          "certIssuerSerial": "",
          "certIssuerDNRFC2253": "",
          "certFingerprintSHA1": ""
        },
        "tlsExportedAuthenticator": {
          "clientFinished": "9dd8cf778d8bf8df67438e9e376247ace5b464b7ce6cee111aa2e4fc5ec63762",
          "clientHandshake": "0cf2123d5162763c637284158a1805ac23c302e92b4c718b1736505fe9bd685b",
          "serverHandshake": "9cd2c03c087a8039231d551fd1405b48aa0edc35c6601449e722918c36af7360",
          "serverFinished": "7bcd72584a49fddd645d5c7e344e3739d1feaa0bc8416a7b08693d9cefc9ca94"
        },
        "tlsVersion": "TLSv1.3",
        "colo": "SIN",
        "timezone": "Asia/Singapore",
        "edgeRequestKeepAliveStatus": 1,
        "requestPriority": "weight=147;exclusive=1",
        "city": "Singapore",
        "httpProtocol": "HTTP/2",
        "asOrganization": "Fdcservers",
        "postalCode": "13"
      }
    }
  },
  "id": 6

**Reply by fucktimi** [reply]
> 你好，再请教下。我是在自己的物理小主机上部署的alist，然后用cloudflared做了内网穿透。然后用workers。然后访问alist的onedrive，看日志会出现canceled的错，你有遇到过吗？
> 
>   "outcome": "canceled",
>   "scriptName": "alist",
>   "exceptions": [],
>   "logs": [],
>   "eventTimestamp": 1664366083185,
>   "event": {
>     "request": {
>       "url": "https://alist.domain.dm/ODmy/%E3%80%8A%E6%AD%A6%E6%9E%97%E5%A4%96%E4%BC%A0%E3%80%8B%E6%9C%AA%E5%88%A0%E8%8A%82%E5%8F%B0%E6%B9%BE%E6%B8%85%E6%99%B0%E7%89%88/[%E6%AD%A6%E6%9E%97%E5%A4%96%E4%BC%A0].02.%E6%8E%8C%E6%9F%9C%E7%9A%84%E6%85%B0%E9%97%AE%E8%A3%85%E7%97%85%E8%80%85.%E8%B7%91%E5%A0%82%E4%BA%BA%E6%94%B6%E6%8B%BE%E5%AF%BB%E8%A1%85%E5%A5%B3.avi?sign=REDACTED=:0",
>       "method": "GET",
>       "headers": {
>         "accept": "*/*",
>         "accept-encoding": "gzip",
>         "accept-language": "zh-CN,zh;q=0.9,en;q=0.8,en-GB;q=0.7,en-US;q=0.6",
>         "cf-connecting-ip": "50.7.252.66",
>         "cf-ipcountry": "SG",
>         "cf-ray": "751c4933ea818820",
>         "cf-visitor": "{\"scheme\":\"https\"}",
>         "connection": "Keep-Alive",
>         "host": "alist.domain.dm",
>         "if-range": "\"{09D21B07-8BD1-451B-8AB8-17665C861D77},5\"",
>         "range": "bytes=1638400-525832191",
>         "sec-ch-ua": "\"Microsoft Edge\";v=\"105\", \"Not)A;Brand\";v=\"8\", \"Chromium\";v=\"105\"",
>         "sec-ch-ua-mobile": "?0",
>         "sec-ch-ua-platform": "\"Windows\"",
>         "sec-fetch-dest": "video",
>         "sec-fetch-mode": "no-cors",
>         "sec-fetch-site": "cross-site",
>         "user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36 Edg/105.0.1343.53",
>         "x-forwarded-proto": "https",
>         "x-real-ip": "50.7.252.66"
>       },
>       "cf": {
>         "clientTcpRtt": 1,
>         "longitude": "103.78360",
>         "latitude": "1.29820",
>         "tlsCipher": "AEAD-AES128-GCM-SHA256",
>         "continent": "AS",
>         "asn": 30058,
>         "clientAcceptEncoding": "identity;q=1, *;q=0",
>         "country": "SG",
>         "tlsClientAuth": {
>           "certIssuerDNLegacy": "",
>           "certIssuerSKI": "",
>           "certSubjectDNRFC2253": "",
>           "certSubjectDNLegacy": "",
>           "certFingerprintSHA256": "",
>           "certNotBefore": "",
>           "certSKI": "",
>           "certSerial": "",
>           "certIssuerDN": "",
>           "certVerified": "NONE",
>           "certNotAfter": "",
>           "certSubjectDN": "",
>           "certPresented": "0",
>           "certRevoked": "0",
>           "certIssuerSerial": "",
>           "certIssuerDNRFC2253": "",
>           "certFingerprintSHA1": ""
>         },
>         "tlsExportedAuthenticator": {
>           "clientFinished": "9dd8cf778d8bf8df67438e9e376247ace5b464b7ce6cee111aa2e4fc5ec63762",
>           "clientHandshake": "0cf2123d5162763c637284158a1805ac23c302e92b4c718b1736505fe9bd685b",
>           "serverHandshake": "9cd2c03c087a8039231d551fd1405b48aa0edc35c6601449e722918c36af7360",
>           "serverFinished": "7bcd72584a49fddd645d5c7e344e3739d1feaa0bc8416a7b08693d9cefc9ca94"
>         },
>         "tlsVersion": "TLSv1.3",
>         "colo": "SIN",
>         "timezone": "Asia/Singapore",
>         "edgeRequestKeepAliveStatus": 1,
>         "requestPriority": "weight=147;exclusive=1",
>         "city": "Singapore",
>         "httpProtocol": "HTTP/2",
>         "asOrganization": "Fdcservers",
>         "postalCode": "13"
>       }
>     }
>   },
>   "id": 6

好像是因为你在workers里面填的alist的地址的域名，跟你路由到workers的那个域名，如果不是属于一个域名下面的子域名的话，就会下载不成功。

**Reply by rakaki** [reply]
两个域名是同一个域名下的。另外想问个问题，我在workers上部署的onedrive反代成功了，能正常访问，但是在onemanager里面使用速度很慢，下载只有几十kb，速度还不如直连，这会是什么原因？

**Reply by fucktimi** [reply]
> 两个域名是同一个域名下的。另外想问个问题，我在workers上部署的onedrive反代成功了，能正常访问，但是在onemanager里面使用速度很慢，下载只有几十kb，速度还不如直连，这会是什么原因？

这我就不清楚了，感觉你的部署很复杂，搞不懂。
